alerter.cpp

```diff
@@ -1,32 +1,37 @@
-#include <iostream>
-#include <assert.h>
+#include "alerter.h"
 
-int alertFailureCount = 0;
+namespace alerter
+{
 
-int networkAlertStub(float celcius) {
-    std::cout << "ALERT: Temperature is " << celcius << " celcius.\n";
-    // Return 200 for ok
-    // Return 500 for not-ok
-    // stub always succeeds and returns 200
-    return 200;
-}
+    void Alerter::alertInCelcius(float farenheit)
+    {
+        float celcius = farenheitToCelcius(farenheit);
+        int returnCode = networkAlertStub(celcius);
+        if (returnCode != 200)
+        {
+            // non-ok response is not an error! Issues happen in life!
+            // let us keep a count of failures to report
+            // However, this code doesn't count failures!
+            // Add a test below to catch this bug. Alter the stub above, if needed.
+            alertFailureCount += 0;
+        }
+    }
 
-void alertInCelcius(float farenheit) {
-    float celcius = (farenheit - 32) * 5 / 9;
-    int returnCode = networkAlertStub(celcius);
-    if (returnCode != 200) {
-        // non-ok response is not an error! Issues happen in life!
-        // let us keep a count of failures to report
-        // However, this code doesn't count failures!
-        // Add a test below to catch this bug. Alter the stub above, if needed.
-        alertFailureCount += 0;
+    float Alerter::farenheitToCelcius(float farenheit)
+    {
+        return (farenheit - 32) * 5 / 9;
     }
 }
 
-int main() {
-    alertInCelcius(400.5);
-    alertInCelcius(303.6);
-    std::cout << alertFailureCount << " alerts failed.\n";
-    std::cout << "All is well (maybe!)\n";
+int main()
+{
+    alerter::Alerter l_alerter;
+    l_alerter.alertInCelcius(400.5);

```

Comment:
stub/test code and production code is the same file. They should be kept separate so that it helps in future maintainability and also test code can be easily excluded during shipment of software to field.